### DISCUSSION TITLE
Why use fold_with(x), while x has already implemented visit_mut

### DISCUSSION THREAD

**Discussion by JSerFeng** [swc-project/swc#4830] [Q&A] [answered]
From source, I found there is one line ```let module = module.fold_with(&mut resolver_with_mark(top_level_mark));```, resolve_with_mark() returns value has already implemented VisitMut, and the doc says that VisitMut is better than fold, am I misunderstanding

**Comment by kdy1** [comment, marked as answer]
`VisitMut` is better, but it's the same for the caller if they call a visitor wrapped with `as_folder`

**Reply by JSerFeng** [reply]
Is calling ```module.visit_mut(&mut resolver_with_mark(top_level_mark))``` directly has problems, thought it is more intuitive

**Reply by kdy1** [reply]
No it's actually better

**Reply by JSerFeng** [reply]
고맙습니다. 늦은 시간에 폐를 끼쳐서 죄송합니다. 안녕히 주무세요.
